### ISSUE TITLE
[Feature Request] AlldebridCom torrents support (iso RealdebridComTorrent)

### ISSUE THREAD

**Issue by Jqh63** [pyload/pyload#3577]
### Adding to AlldebridCom plugin a torrents support through alldebrid API

Can you please implement same configuration made for RealdebridComTorrent but using alldebrid

- Target is to use hotfolder as blackhole for .torrent file input, send them to alldebrid API and get download links to queue. 

### Additional
- torrents can ref multiple files, filefilter before transfert to download queue is a good option

Tried on my own from other py ref but hard to include in a downloader.
Plus, now alldebrid manage a nice and simple API for all that stuff, I am just missing automation.
I can provide alldebrid account for testing.

### Additional references
[alldebrid API](https://docs.alldebrid.com/v4/#magnet)

Ref to [#3223](https://github.com/pyload/pyload/issues/3223)

Old code ref [DanySK/torrent2magnet](https://github.com/DanySK/torrent2magnet/blob/develop/torrent2magnet.py) [BloQster/docker-magnet-alldebrid-jd](https://github.com/BloQster/docker-magnet-alldebrid-jd/blob/master/magnet-alldebrid-jd.py)

**Comment by GammaC0de** [comment]
Account information is needed for testing, please set to a temporary password and then send account user and password to nitzo2001 at yahoo dot com.

**Comment by Jqh63** [comment]
done.
Plus, I notice alldebrid login into pyLoad only work with the key provide using your external script and not a token generated using website. Is it a bug? maybe another issue to create

**Comment by GammaC0de** [comment]
OK, Finally I finished it.

Please note that because of the conflict with `RealDebridTorrent` plugin, to be able to use the torrent feature, you will have to to edit [AlldebridComTorrent.py#L23](https://github.com/pyload/pyload/blob/dd3d6108fb8001c1ff93a39b54539f5dc9cb704d/module/plugins/crypter/AlldebridComTorrent.py#L23):
Change:
```python
    #_pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
    __pattern__ = r'^unmatchable$'
```
To:
```python
    __pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
```
Also you have to change [RealdebridComTorrent.py#L23](https://github.com/pyload/pyload/blob/master/src/pyload/plugins/downloaders/RealdebridComTorrent.py#L23) from:
```python
    __pattern__ = r"(?:file|https?)://.+\.torrent|magnet:\?.+"
```
To:
```python
    __pattern__ = r'^unmatchable$'
```

This is because pyLoad does not support deactivating hoster and crypter plugins and adding this feature would require a new pyLoad release.
You would have to do these changes on every update of `RealdebridComTorrent ` and`AlldebridComTorrent`, sorry.

Also, one more thing:
because of the way Realdebrid accepts file uploads, to be able to send a torrent file to `RealdebridComTorrent`, you would have to update your pyload installation using the instructions in #3283 so `HTTPRequest` gets updated too.

Please tell if it works or not.

**Comment by lukastribus** [comment]
@GammaC0de I noticed "Error calling event handler plugin_updated" warnings:

```
17.04.2020 00:08:20	INFO	ADDON UpdateManager: pyLoad is up to date!
17.04.2020 00:08:21	INFO	ADDON UpdateManager: New version of plugin: ACCOUNT AlldebridCom (v0.43 -> v0.44)
17.04.2020 00:08:21	INFO	ADDON UpdateManager: New plugin: CRYPTER AlldebridComTorrent (v0.01)
17.04.2020 00:08:21	INFO	ADDON UpdateManager: New version of plugin: HOSTER AlldebridCom (v0.59 -> v0.60)
17.04.2020 00:08:22	INFO	ADDON UpdateManager: *** Plugins updated ***
17.04.2020 00:08:25	WARNING	Error calling event handler plugin_updated: >, ([(u'accounts', u'AlldebridCom'), (u'crypter', u'AlldebridComTorrent'), (u'hoster', u'AlldebridCom')],), 'validuntil'
17.04.2020 00:08:25	WARNING	Error calling event handler plugin_updated: >, ([(u'accounts', u'AlldebridCom'), (u'crypter', u'AlldebridComTorrent'), (u'hoster', u'AlldebridCom')],), 'data' 
```


Not sure if there is any impact, just wanted to let you know.

**Comment by Jqh63** [comment]
From my first try using a fresh docker image with pyload stable branch and your mods into crypter files:
- now token generated from Alldebrid website is working for an account add and no more using the token from your py script (zip), not sure why but working anyway
- I manually disable updatemanager plugin and other conflicts plugin like XFileSharing and UptoboxCom (should be disable by default I guess)
- configured hotfolder connect to blackhole from my media managers and everything is working fine 👍 
- tests made: .torrent file, .magnet file, UTF-8 accent, mulltiples kind of files, already upload or not as direct link from alldebrid torrent manager, simultaneously inputs, ...

### Good job!

Actual bugs found: 
- .magnet files don't trigger Alldebrid plugin
- accent issue with file name using é or è. Error message: `'ascii' codec can't encode character u'\xe9' in position 35: ordinal not in range(128)`

Improvment:
- hotfolder scan refresh time
- file filters to get from Alldebrid like only *mkv, *avi, *deb
 
Plus I got another point, how pyload manage all old packages into queue list, as it will become a huge list with many missing files. How can I cleaning it the right way?

**Comment by GammaC0de** [comment]
@Jqh63 
>   .magnet files don't trigger Alldebrid plugin

So what plugin does it trigger?

>  accent issue with file name using é or è. Error message: 'ascii' codec can't encode character u'\xe9' in position 35: ordinal not in range(128)

Debug log please,
Can you point me to the torrent file?

> Plus I got another point, how pyload manage all old packages into queue list, as it will become a huge list with many missing files. How can I cleaning it the right way?

You press the clear finished button in the queue tab (the little recycle bin button)

@lukastribus 
> I noticed "Error calling event handler plugin_updated" warnings:

I noticed this to, it happens only after UpdateManager finishes updating.
If you restart pyload, you will not see it again.
I have to investigate this issue.

**Comment by Jqh63** [comment]
here is log of both issues (from random sources) : 
- magnet - it tried to trigger baseplugin without success and same message even if I disable it
- special chars - french sources in this case

### [pyload_log_ascii_magnet.txt](https://github.com/pyload/pyload/files/4493748/pyload_log_ascii_magnet.txt)

**Comment by GammaC0de** [comment]
> .magnet files don't trigger Alldebrid plugin

magnets are URLs, not files...

What are those .magnet files you are trying to download?

> 'ascii' codec can't encode character u'\xe9' in position 18: ordinal not in range(128)

This is actually (and sadly) a pycurl bug, I will try to update `HTTPRequest.py` to include a workaround this bug.

**Comment by Jqh63** [comment]
Hope you will find a way to solve the special chars issue.
About magnet file, in fact some indexers provide a `.magnet` file instead of a `.torrent` file.
Like:
`<torrentname>.magnet`
and with content like:
`magnet:?xt=urn:btih:1fcb0d10c3b6840246dbcc398ad77f5149bea699&dn=The+New+York+Times+Best+Sellers+Fiction+-+March+24%2C+2019&xl=22653273&tr=udp%3A%2F%2Ftracker.coppersurfer.tk:6969/announce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org:6969/announce&tr=udp%3A%2F%2Ftracker.pirateparty.gr:6969/announce&tr=udp%3A%2F%2Ftracker.internetwarriors.net:1337/announce&tr=udp%3A%2F%2Fopen.stealth.si:80/announce&tr=udp%3A%2F%2Fexplodie.org:6969/announce&tr=udp%3A%2F%2Ftracker.opentrackr.org:1337/announce&tr=udp%3A%2F%2Ftracker.tiny-vps.com:6969/announce&tr=udp%3A%2F%2Fopen.demonii.si:1337/announce&tr=udp%3A%2F%2Ftracker.torrent.eu.org:451/announce&tr=udp%3A%2F%2Fbt.xxx-tracker.com:2710/announce&tr=udp%3A%2F%2Fipv4.tracker.harry.lu:80/announce&tr=udp%3A%2F%2Ftracker.cyberia.is:6969/announce&tr=udp%3A%2F%2F9.rarbg.to:2710/announce&tr=udp%3A%2F%2Fdenis.stalker.upeer.me:6969/announce`

So typically just the magnet URL into a file.
Hence pyLoad can handle magnet URL as package input but not into a file I guess. Should I create a new related feature request for hotfolder ?

**Comment by GammaC0de** [comment]
`HTTPRequest.py ` updated, should work now.

Where can i get .magnet files to test?

**Comment by Jqh63** [comment]
sent one by email: public tracker and simple file

**Comment by GammaC0de** [comment]
magnet files now works

**Comment by Jqh63** [comment]
Both issues have been solved.
Nice 👍 and thank you. 
I will continue to test during the following days anyway.

**Comment by GammaC0de** [comment]
Added check interval configuration option to `HotFolder`

**Comment by Jqh63** [comment]
Got this error from log using last HotFolder:

442 | 18.04.2020 22:55:08 | INFO | pyLoad is up and running
-- | -- | -- | --
443 | 18.04.2020 22:55:10 | ERROR | <<< UNCAUGHT EXCEPTION >>>
444 | ? | ? | Traceback (most recent call last):
445 | ? | ? | File "/app/pyload/module/Scheduler.py", line 101, in run
446 | ? | ? | ret = self.call(*self.args, **self.kwargs)
447 | ? | ? | File "/app/pyload/module/plugins/internal/misc.py", line 185, in _task
448 | ? | ? | self.restart(threaded=threaded, delay=self.interval)
449 | ? | ? | File "/app/pyload/module/plugins/internal/misc.py", line 163, in restart
450 | ? | ? | return self.start(*args, **kwargs)
451 | ? | ? | File "/app/pyload/module/plugins/internal/misc.py", line 158, in start
452 | ? | ? | self.cb = self.plugin.pyload.scheduler.addJob(max(1, delay), self._task, [threaded], threaded=threaded)
453 | ? | ? | File "/app/pyload/module/Scheduler.py", line 55, in addJob
454 | ? | ? | t += time()
455 | ? | ? | TypeError: coercing to Unicode: need string or buffer, float found

Not sure if I missed something, I will check tomorrow

**Comment by GammaC0de** [comment]
Oopsi 🤦‍♂️ my mistake,
Fixed.

**Comment by Jqh63** [comment]
Working fine now. Nice.
Plus another info, from my other load tests, I have to reduce connect per package and concurrents downloads to 2 on both settings to avoid following error: 
119 | 18.04.2020 17:37:36 | INFO | CRYPTER AlldebridComTorrent[134]: Save package `filename` to folder: filename
-- | -- | -- | --
120 | 18.04.2020 17:37:51 | WARNING | Download failed: filename \| Downloaded content was smaller than expected. Try to reduce download connections

Seems an Alldebrid or uptobox DDoS protection.
Another point, but due to bad tracker uploader I think, package use naming from tracker announce and not from .torrent filename. Sometimes information are missing like revision of file to download.
Maybe a new settings to implement to use torrent file actual filename and not the one trom tracker announce.

**Comment by GammaC0de** [comment]
> Seems an Alldebrid or uptobox DDoS protection.
Another point, but due to bad tracker uploader I think, package use naming from tracker announce and not from .torrent filename. Sometimes information are missing like revision of file to download.
Maybe a new settings to implement to use torrent file actual filename and not the one trom tracker announce.

Please back up all the issues with debug logs so I can understand whats happening

**Comment by ManlyJack** [comment]
I have a docker image installed, browsing to the config I am only able to find the alldebridcomtorrent.py file to edit and cannot find the realdebridcomtorrent.py file anywhere at all, I edited the alldebrid file and it still throws the realdebrid premium account required error.
can you point me in the right direction here? 
Unraid pyload docker

**Comment by Jqh63** [comment]
@GammaC0de no real issue because not blocking, I will continue to experience this plugin
@ManlyJack which docker image are you using? clean one is [linuxserver/pyload](https://hub.docker.com/r/linuxserver/pyload)

**Comment by ManlyJack** [comment]
Yes the linuxserver one, I seem to have found them, they aren't in the appdata folder but within the docker itself.  had to edit them via the console for the docker.  The files I edited are in /app/pyload/module/plugins/crypter and it looks as though it is working but it was trying to use XFileSharing to download them so had to disable this in the pyload config also.

**Comment by GammaC0de** [comment]
👍 
>  but it was trying to use XFileSharing to download 


Yes, `XFileSharing `has this bad habit.

**Comment by Miosame** [comment]
@GammaC0de if I don't need realdebrid(torrent), is there a way to permanently remove it so alldebridtorrent plugin would work without doing this on all updates: https://github.com/pyload/pyload/issues/3577#issuecomment-614912141?

**Comment by Miosame** [comment]
The changes also don't seem to have any effect on pyload detecting magnets as realdebrid. I made sure multiple times that the changes are in fact inside the container and reloaded pyload, but it still forces realdebrid for magnets, is there anything else one has to do?

**Comment by TuXbOy06** [comment]
Hi, I did all the modifications above 

Please note that because of the conflict with RealDebridTorrent plugin, to be able to use the torrent feature, you will have to to edit AlldebridComTorrent.py#L23:
Change:

    #_pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
    __pattern__ = r'^unmatchable$'
To:

    __pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
Also you have to change RealdebridComTorrent.py#L23 from:

    __pattern__ = r"(?:file|https?)://.+\.torrent|magnet:\?.+"
To:

    __pattern__ = r'^unmatchable$'

But Pyload is still loading Realdebridtorrent plugin... 


23.04.2020 14:42:23 | ERROR | Decrypting failed: urnbtih361156cca81870f5bd508dffcdc74bf346e760a8 \| This plugin requires an active Realdebrid.com account
-- | -- | --
379 | 23.04.2020 14:42:23 | INFO | CRYPTER RealdebridComTorrent[2]: Processing url: magnet:?xt=urn:btih:361156cca81870f5bd508dffcdc74bf346e760a8&dn=Full.Out.2.You.Got.This.2020.WEB-DL.x264-FGT&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2740&tr=udp%3A%2F%2F9.rarbg.to%3A2870
378 | 23.04.2020 14:42:23 | DEBUG | ADDON UserAgentSwitcher: Use custom user-agent string `Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:71.0) Gecko/20100101 Firefox/71.0`
377 | 23.04.2020 14:42:23 | DEBUG | ADDON UserAgentSwitcher: Setting maximum redirections to 10
376 | 23.04.2020 14:42:23 | DEBUG | ADDON UserAgentSwitcher: Setting connection timeout to 60 seconds
375 | 23.04.2020 14:42:23 | DEBUG | ADDON ExternalScripts: No script found under folder `download_preparing`
374 | 23.04.2020 14:42:23 | INFO | CRYPTER RealdebridComTorrent[2]: Link status: starting
373 | 23.04.2020 14:42:23 | INFO | CRYPTER RealdebridComTorrent[2]: Link size: N/D
372 | 23.04.2020 14:42:23 | INFO | CRYPTER RealdebridComTorrent[2]: Link name: urnbtih361156cca81870f5bd508dffcdc74bf346e760a8
371 | 23.04.2020 14:42:23 | DEBUG | CRYPTER RealdebridComTorrent[2]: Previous link info: {}
370 | 23.04.2020 14:42:23 | DEBUG | CRYPTER RealdebridComTorrent[2]: Link info: {'status': 7, 'hash': {}, 'name': u'urnbtih361156cca81870f5bd508dffcdc74bf346e760a8', 'url': u'magnet:?xt=urn:btih:361156cca81870f5bd508dffcdc74bf346e760a8&dn=Full.Out.2.You.Got.This.2020.WEB-DL.x264-FGT&tr=http://tracker.trackerfix.com:80/announce&tr=udp://9.rarbg.me:2740&tr=udp://9.rarbg.to:2870', 'pattern': {}, 'size': 0}
369 | 23.04.2020 14:42:23 | INFO | CRYPTER RealdebridComTorrent[2]: Grabbing link info...
368 | 23.04.2020 14:42:23 | DEBUG | CRYPTER RealdebridComTorrent[2]: Config option `use_premium` not found, use default `True`
367 | 23.04.2020 14:42:23 | WARNING | CRYPTER RealdebridComTorrent[2]: Plugin may be unstable

I noticed that after adding a magnet link Pyload is telling right away 

![image](https://user-images.githubusercontent.com/13116073/80101121-4265fc00-8571-11ea-9887-be67bbc709e9.png)

I don't even have Realdebridtorrent in my list to disable it

![image](https://user-images.githubusercontent.com/13116073/80101211-688b9c00-8571-11ea-9a66-576e2448dfb2.png)

Can someone help? I am using Pyload on a docker, I am using writl/pyload with a Rancher ( Docker orchestrator)
Thanks

**Comment by GammaC0de** [comment]
To block `RealdebridComTorrent.py` from updating and make `AlldebridComTorrent.py` handle magnets and torrents, do this:
- Go to `http://<PYLOAD_IP>/info` and write down the value of `Config Folder`.
- Stop pyLoad.
- Download https://raw.githubusercontent.com/pyload/pyload/stable/module/plugins/crypter/RealdebridComTorrent.py to  `<Config Folder>/userplugins/RealdebridComTorrent.py` directory.
- Edit `<Config Folder>/userplugins/RealdebridComTorrent.py` and:
  * Change `__version__` to `2.00`
  * Change [RealdebridComTorrent.py#L23](https://github.com/pyload/pyload/blob/master/src/pyload/plugins/downloaders/RealdebridComTorrent.py#L23) from:
```python
    __pattern__ = r"(?:file|https?)://.+\.torrent|magnet:\?.+"
```
To:
```python
    __pattern__ = r'^unmatchable$'
```
- Edit `<Config Folder>/userplugins/AlldebridComTorrent.py` and:
   * Change [AlldebridComTorrent.py#L23](https://github.com/pyload/pyload/blob/dd3d6108fb8001c1ff93a39b54539f5dc9cb704d/module/plugins/crypter/AlldebridComTorrent.py#L23) from:

```python
    #_pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
    __pattern__ = r'^unmatchable$'
```
To:
```python
    __pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
```
i.e. uncomment the first line and delete the second line.

- Start pyLoad

**Comment by TuXbOy06** [comment]
Ok just did it but doesn't work 

23.04.2020 16:57:37	ERROR	Decrypting failed: urnbtih361156cca81870f5bd508dffcdc74bf346e760a8 | This plugin requires an active Realdebrid.com account
10	23.04.2020 16:57:37	INFO	CRYPTER RealdebridComTorrent[2]: Processing url: magnet:?xt=urn:btih:361156cca81870f5bd508dffcdc74bf346e760a8&dn=Full.Out.2.You.Got.This.2020.WEB-DL.x264-FGT&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2740&tr=udp%3A%2F%2F9.rarbg.to%3A2870
9	23.04.2020 16:57:37	DEBUG	ADDON UserAgentSwitcher: Use custom user-agent string `Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:71.0) Gecko/20100101 Firefox/71.0`
8	23.04.2020 16:57:37	DEBUG	ADDON UserAgentSwitcher: Setting maximum redirections to 10
7	23.04.2020 16:57:37	DEBUG	ADDON UserAgentSwitcher: Setting connection timeout to 60 seconds
6	23.04.2020 16:57:37	DEBUG	ADDON ExternalScripts: No script found under folder `download_preparing`
5	23.04.2020 16:57:37	INFO	CRYPTER RealdebridComTorrent[2]: Link status: starting
4	23.04.2020 16:57:37	INFO	CRYPTER RealdebridComTorrent[2]: Link size: N/D
3	23.04.2020 16:57:37	INFO	CRYPTER RealdebridComTorrent[2]: Link name: urnbtih361156cca81870f5bd508dffcdc74bf346e760a8
2	23.04.2020 16:57:37	DEBUG	CRYPTER RealdebridComTorrent[2]: Previous link info: {}
1	23.04.2020 16:57:37	DEBUG	CRYPTER RealdebridComTorrent[2]: Link info: {'status': 7, 'hash': {}, 'name': u'urnbtih361156cca81870f5bd508dffcdc74bf346e760a8', 'url': u'magnet:?xt=urn:btih:361156cca81870f5bd508dffcdc74bf346e760a8&dn=xxxxxxxxxx.xxxx.WEB-DL.x264-FGT&tr=http://tracker.trackerfix.com:80/announce&tr=udp://9.rarbg.me:2740&tr=udp://9.rarbg.to:2870', 'pattern': {}, 'size': 0}

Here a copy of the folder config http://www.mediafire.com/file/isfj9rg27f9dtdr/pyload_config.tar/file

I removed my login informations for Alldebrid, if you want a login to test i will send you by mail... Please let me know

**Comment by Miosame** [comment]
@GammaC0de I see `file:` being matched against, is there a way to select `.torrent` files when adding to queue or do you have to craft those manually for now?

**Comment by GammaC0de** [comment]
@TuXbOy06 Are you retrying the same existing package?
Please try creating a new package

**Comment by TuXbOy06** [comment]
@GammaC0de which package are you talking about the .tar I created for you? or a new magnet link?

**Comment by GammaC0de** [comment]
new magnet link

**Comment by TuXbOy06** [comment]
@GammaC0de man you rock its working thank you very much

**Comment by TuXbOy06** [comment]
Ok solved it by disable UptoboxCom plugin and by forcing uptobox in AlldebridCom plugin

![image](https://user-images.githubusercontent.com/13116073/80218852-ffc02480-8641-11ea-81bb-ff985cadc0a4.png)

**Comment by ManlyJack** [comment]
> 
> To block `[RealdebridComTorrent.py](http://RealdebridComTorrent.py)` from updating and make `[AlldebridComTorrent.py](http://AlldebridComTorrent.py)` handle magnets and torrents, do this:
> 
>     * Go to `http://<PYLOAD_IP>/info` and write down the value of `Config Folder`.
> 
>     * Stop pyLoad.
> 
>     * Download https://raw.githubusercontent.com/pyload/pyload/stable/module/plugins/crypter/RealdebridComTorrent.py to  `<Config Folder>/userplugins/[RealdebridComTorrent.py](http://RealdebridComTorrent.py)` directory.
> 
>     * Edit `<Config Folder>/userplugins/[RealdebridComTorrent.py](http://RealdebridComTorrent.py)` and:
>       
>       * Change `__version__` to `2.00`
>       * Change [RealdebridComTorrent.py#L23](https://github.com/pyload/pyload/blob/master/src/pyload/plugins/downloaders/RealdebridComTorrent.py#L23) from:
> 
> 
> ```python
>     __pattern__ = r"(?:file|https?)://.+\.torrent|magnet:\?.+"
> ```
> 
> To:
> 
> ```python
>     __pattern__ = r'^unmatchable$'
> ```
> 
>     * Edit `<Config Folder>/userplugins/[AlldebridComTorrent.py](http://AlldebridComTorrent.py)` and:
>       
>       * Change [AlldebridComTorrent.py#L23](https://github.com/pyload/pyload/blob/dd3d6108fb8001c1ff93a39b54539f5dc9cb704d/module/plugins/crypter/AlldebridComTorrent.py#L23) from:
> 
> 
> ```python
>     #_pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
>     __pattern__ = r'^unmatchable$'
> ```
> 
> To:
> 
> ```python
>     __pattern__ = r'(?:file|https?)://.+\.torrent|magnet:\?.+'
> ```
> 
> i.e. uncomment the first line and delete the second line.
> 
>     * Start pyLoad

If I put these files in the userplugins directory do they override the ones in the standard config in the docker? or will I still have to edit the files with each update? 
It just updated on me and reset it all haha

**Comment by GammaC0de** [comment]
> If I put these files in the userplugins directory do they override the ones in the standard config in the docker? 

Yes, pyload uses the highest version of of either one of the plugins.

> or will I still have to edit the files with each update?

If you don't update the docker image, there should not be a problem

> It just updated on me and reset it all haha

Sorry no understand

**Comment by ManlyJack** [comment]
> 
> 
> > If I put these files in the userplugins directory do they override the ones in the standard config in the docker?
> 
> Yes, pyload uses the highest version of of either one of the plugins.
> 
> > or will I still have to edit the files with each update?
> 
> If you don't update the docker image, there should not be a problem

On a fresh update I created and edited the two files in the userplugins directory and it didn't seem to work, I'll have to try again when I get another update,  but I have disabled the auto-update for now anyway.
 
> > It just updated on me and reset it all haha
> 
> Sorry no understand

Pyload docker auto-updated and lost all my customisations, hence having to re-do them all.

Thanks for all your help with this, much appreciated!

**Comment by Miosame** [comment]
@ManlyJack you can mount overwrite these in docker-compose or via `-v` on docker run, here's an example for docker-compose:

```
      - "./pyload_modules/AlldebridComTorrent.py:/app/pyload/module/plugins/crypter/AlldebridComTorrent.py:ro"
      - "./pyload_modules/RealdebridComTorrent.py:/app/pyload/module/plugins/crypter/RealdebridComTorrent.py:ro"
      - "./pyload_modules/RealdebridComTorrent.py:/config/userplugins/crypter/RealdebridComTorrent.py:ro"
```

note that I am overwriting all occurences of realdebrid, which for me were also in userplugins.

**Comment by ManlyJack** [comment]
@Miosame Thanks, I'm not familiar with docker-compose and am running docker on an Unraid box, is there a walkthrough video or doc somewhere around I could have a read up about docker-compose? if this is even possible on Unraid?

**Comment by Miosame** [comment]
@ManlyJack I actually do have unraid too, but sadly no access to it right now, so can't look around / test. You basically want to find a way to mount volumes just like I did above - in the unraid docker web-ui when you go and edit the container, this sounds promising: https://wiki.unraid.net/UnRAID_6/Docker_Management (Volume Mappings) and this looks promising: https://wiki.unraid.net/What_are_the_host_volume_paths_and_the_container_paths basically adding from the host filesystem those `AlldebridComTorrent.py` and `RealdebridComTorrent.py` into the container.

**Comment by ManlyJack** [comment]
@Miosame I think the host volume mapping only works the other way, however have never tested it the other way.  

The userplugins directory is stored in the appdata share on unraid which is accessible directly anyway but when I added and edited the two files to that directory it didn't seem to work.

@GammaC0de said it uses the highest version of these plugins files from either directory so maybe that has something to do with it not choosing the edited versions? how can I force the userplugins file to be a higher version than the other?

**Comment by GammaC0de** [comment]
> how can I force the userplugins file to be a higher version than the other?

Just edit the file and set `__version__ ` as desired.

**Comment by ManlyJack** [comment]
@GammaC0de  awesome, thanks.  I'll have to try that if the config gets reset again.

**Comment by GammaC0de** [comment]
OK, `RealdebridComTorrent ` was just updated so you all have a chance to test if the changes you have made are resilient to this  :grimacing:

**Comment by Miosame** [comment]
lol thanks, mine worked, though now I should probably diff-check what changed and merge it too

**Comment by GammaC0de** [comment]
For those who haven't succeed yes, I have found a mistake in my instructions in https://github.com/pyload/pyload/issues/3577#issuecomment-618438008 and updated it (mainly, the destination directory of the .py file was wrong).

@Jqh63 
> Improvment:
    * hotfolder scan refresh time
    * file filters to get from Alldebrid like only *mkv, *avi, *deb

Sadly, Alldebrid does not support selecting which files to download from torrents, Realdebrid does.

**Comment by Miosame** [comment]
@GammaC0de why does the alldebrid plugin link to a specific commit blob instead of the stable branch like realdebrid? on stable there's also still the comment that could lure people into uncommenting it wrong: https://github.com/pyload/pyload/blob/stable/module/plugins/crypter/AlldebridComTorrent.py#L23

Since it assigns to `_pattern__` instead of `__pattern__` - the linked blob has it already removed, but wondering why it's not on stable?

**Comment by GammaC0de** [comment]
I wanted to link to the current version of the file in case there are changes in the future.

**Comment by GammaC0de** [comment]
https://github.com/pyload/pyload/commit/6d0ccf638712ae01ff8aabce30aed68a86dfa421 Now there is a way to choose between `RealdebridComTorrent` and `AlldebridComTorrent`. so everyone please update your plugins.

**Comment by Miosame** [comment]
@GammaC0de so we don't need to overwrite anymore, can just pull the updated versions always? thanks for the work btw

**Comment by GammaC0de** [comment]
Yes, Just go to plugin settings -> TORRENT and select desired plugin
 
 
  Miosame<notifications@github.com> wrote:   


@GammaC0de so we don't need to overwrite anymore, can just pull the updated versions always? thanks for the work btw

**Comment by ManlyJack** [comment]
> 
> Yes, Just go to plugin settings -> TORRENT and select desired plugin 

I don't have an option to choose plugin? just activated, create folder and use premium?

**Comment by ManlyJack** [comment]
Disregard! it is showing up now!

**Comment by ManlyJack** [comment]
I keep getting a "No plugin is associated with torrents / magnets" error, I have removed and reinstalled the docker, and cleaned out the userplugins, I'll try completely blowing the whole install away and reconfigure from scratch and see if that helps.

**Comment by GammaC0de** [comment]
This is message is intented. 
See instructions above
 
 
  
<notifications@github.com> wrote:   


I keep getting a "No plugin is associated with torrents / magnets" error, I have removed and reinstalled the docker, and cleaned out the userplugins, I'll try completely blowing the whole install away and reconfigure from scratch and see if that helps.

**Comment by GammaC0de** [comment]
Here: https://github.com/pyload/pyload/issues/3577#issuecomment-626255257
 
Miosame<notifications@github.com> wrote:   


lol thanks, mine worked, though now I should probably diff-check what changed and merge it too

**Comment by ManlyJack** [comment]
Have set it to use alldebrid multiple times, saved, still get the same thing, restart the docker and the setting has reverted back to "None" instead of Alldebrid
 
> This is message is intented.  See instructions above

**Comment by GammaC0de** [comment]
Tell me if it works without reading 

 
 
  ManlyJack<notifications@github.com> wrote:   


Have set it to use alldebrid multiple times, saved, still get the same thing, restart the docker and the setting has reverted back to "None" instead of Alldebrid


This is message is intented. See instructions above

**Comment by ManlyJack** [comment]
> 
> Tell me if it works without reading  ManlyJack<notifications@github.com> wrote: Have set it 

I'm assuming you mean restarting, 
Fresh restart, set the setting to alldebrid, load a magnet file and still get the same, it says saved successfully when hitting the save button

**Comment by ManlyJack** [comment]
Here is the tail end of the log after a restart of pyload, setting the setting, loading a magnet

10.05.2020 13:48:41 INFO ADDON TORRENT: Using AlldebridComTorrent to handle torrents / magnets
10.05.2020 13:49:01 INFO Added package add_file containing 1 links
10.05.2020 13:49:02 INFO Decrypting starts: /downloads/tmp_tmp_Empty.magnet
10.05.2020 13:49:02 WARNING CRYPTER TORRENT[1]: Plugin may be unstable
10.05.2020 13:49:02 INFO CRYPTER TORRENT[1]: Grabbing link info...
10.05.2020 13:49:02 INFO CRYPTER TORRENT[1]: Link name: tmp_tmp_Empty.magnet
10.05.2020 13:49:02 INFO CRYPTER TORRENT[1]: Link size: N/D
10.05.2020 13:49:02 INFO CRYPTER TORRENT[1]: Link status: starting
10.05.2020 13:49:02 INFO CRYPTER TORRENT[1]: Processing url: /downloads/tmp_tmp_Empty.magnet
10.05.2020 13:49:02 ERROR CRYPTER TORRENT[1]: No plugin is associated with torrents / magnets | Please go to plugin settings -> TORRENT and select your preferred plugin
10.05.2020 13:49:02 ERROR Decrypting failed: tmp_tmp_Empty.magnet | No plugin is associated with torrents / magnets

**Comment by ManlyJack** [comment]
Am getting this error in the logs on bootup of a complete fresh install

10.05.2020 14:19:01 WARNING ADDON UpdateManager: Error removing `/app/pyload/module/plugins/accounts/GetTwentyFourOrg.py` | [Errno 13] Permission denied: '/app/pyload/module/plugins/accounts/GetTwentyFourOrg.py'
10.05.2020 14:19:01 WARNING ADDON UpdateManager: Error removing `/app/pyload/module/plugins/hoster/GetTwentyFourOrg.py` | [Errno 13] Permission denied: '/app/pyload/module/plugins/hoster/GetTwentyFourOrg.py'

Could this mean Pyload is having issues with premissions on files/folders? It saves other settings correctly but the TORRENT one doesn't stick or seem to have any effect.

**Comment by TuXbOy06** [comment]
@GammaC0de thanks work like charm

**Comment by ManlyJack** [comment]
Still cannot get this to work, have completely removed pyload and all config files, reinstalled fresh, setting the TORRENT plugin to use anything says it has made the change in the logs, but it doesn't work, and after restarting Pyload it throws a "torrents/magnets are not assigned to any plugin" in the logs and the config has returned to "None".

**Comment by TuXbOy06** [comment]
> Still cannot get this to work, have completely removed pyload and all config files, reinstalled fresh, setting the TORRENT plugin to use anything says it has made the change in the logs, but it doesn't work, and after restarting Pyload it throws a "torrents/magnets are not assigned to any plugin" in the logs and the config has returned to "None".
> 
> Same exact issue on 2 machines running Pyload in Docker, now neither can handle torrents at all. Is there a delay in this being pushed out completely to the docker version? I thought the linuxserver/pyload was pulling direct from the git?

Hi try to pull this docker from the hub writl/pyload this is one I am using...

**Comment by ManlyJack** [comment]
> 
> Hi try to pull this docker from the hub writl/pyload this is the one I am using...

It appears to be an issue not related to Docker, I installed on windows and also have the exact same thing happen. the settings for the TORRENT plugin inside the plugin.conf file get reset to NONE on every restart of pyload.

**Comment by GammaC0de** [comment]
Plugins updated, please check

**Comment by ManlyJack** [comment]
> 
> 
> Plugins updated, please check

Works perfect! thanks for all the help through this @GammaC0de

**Comment by GammaC0de** [comment]
:+1:
